Source/Provers/SMTLib/SMTLibLineariser.cs

```diff
@@ -26,31 +26,50 @@ public class LineariserOptions
   // Lineariser for expressions. The result (bool) is currently not used for anything
   public class SMTLibExprLineariser : IVCExprVisitor<bool, LineariserOptions /*!*/>
   {
-    public static string StoreOpName(VCExprNAry node)
+    public SMTLibOptions LibOptions { get; }
```

Comment:
Is it bad style to use SMTLibOptions for the name of the field also?

Reply:
I don't think so. I would have preferred to name it `Options` but that was often already taken by a field containing `ProverOptions`. I think ideally `SMTLibOptions` and `ProverOptions` would be related somehow and then we'd only need a single `Options` field, but I've left that out of scope for this PR.

Reply:
@bkragl :  FYI.  Your input about ProverOptions and SMTLibOptions and potential unification would be valuable since you spent a ton of time earlier cleaning up the Boogie-Prover interface.